haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellRewriter.java

```diff
@@ -112,18 +120,18 @@ public RewriterResult execute(K k, Optional<Integer> depth) {
                         "--module", moduleName,
                         "--pattern", pgmPath,
                         "--output", koreOutputFile.getAbsolutePath()));
-                if (options.depth != null) {
+                if (depth.isPresent()) {
                     args.add("--depth");
-                    args.add(options.depth.toString());
+                    args.add(depth.toString());
```

Comment:
the java library specification does not provide any guarantees about the value of Optional.toString so you really should be calling get() here.

Reply:
Fixed.